### PR TITLE
JPDA and SystematicResampler Fixes (updated)

### PR DESCRIPTION
**Pull request by sglvladi**
This PR supersedes #152. The following changes have been applied:
- The branch has been rebased to the latest ``master`` commit.
- An ``attrgetter`` is now used when sorting the weights in ``SystematicResampler``
- Minor formatting has been applied to ``JPDA``.

### CODE REVIEW

File: stonesoup/resampler/particle.py [hunk 1]
```diff
@@ -24,6 +25,7 @@ def resample(self, particles):
 
         n_particles = len(particles)
         weight = Probability(1/n_particles)
+        particles = sorted(particles, key=attrgetter('weight'), reverse=False)
         cdf = np.cumsum([p.weight for p in particles])
         particles_listed = list(particles)
```
Comment (sdhiscocks) [inline stonesoup/resampler/particle.py:30]: Just realised that as `particles` is sorted on line 28 (R), that it will already be a list.

Comment (sglvladi) [reply stonesoup/resampler/particle.py]: Well spotted 👍 That's been corrected now.

**Comment by sdhiscocks** [comment]
@sglvladi - Could you please rebase this to drop the delete commits, and squash duplicate commits.

**Comment by sglvladi** [comment]
> @sglvladi - Could you please rebase this to drop the delete commits, and squash duplicate commits.

@sdhiscocks I've had a go at it. Let me know if there's anything else that needs doing.

**Comment by sdhiscocks** [comment]
Thanks @sglvladi

**Pull request merged**